charts/dvpe-deployment-gloo/templates/secret.yaml

```diff
@@ -2,58 +2,65 @@
-  backendType: secretsManager
-  {{- if .Values.externalSecrets.service.roleArn }}
-  roleArn: {{ .Values.externalSecrets.service.roleArn }}
-  {{- end}}
+  secretStoreRef:
+    kind: SecretStore
+    name: external-secret-store-{{ .Release.Namespace }}
+  target:
+    name: {{ $serviceName }}-service-secrets
   dataFrom:
-    - {{ .Values.externalSecrets.service.key }}
+  - extract:
+      key: {{ .Values.externalSecrets.service.key }}
+  refreshInterval: {{ .Values.externalSecrets.service.refreshInterval | default "15m" }}
 {{- end }}
 
 {{- if and .Values.externalSecrets.oauth2.key .Values.gloo.authConfig.spec.configs.oauth.clientId }}
 ---
-apiVersion: 'kubernetes-client.io/v1'
+apiVersion: 'external-secrets.io/v1beta1'
 kind: ExternalSecret
 metadata:
   name: {{ $serviceName }}-oidc-secrets
   namespace: {{ .Release.Namespace }}
 spec:
-  backendType: secretsManager
-  {{- if .Values.externalSecrets.service.roleArn }}
-  roleArn: {{ .Values.externalSecrets.service.roleArn }}
-  {{- end}}
-  template:
-    metadata:
-      annotations:
-        resource_kind: '*v1.Secret'
+  secretStoreRef:
+    kind: SecretStore
+    name: external-secret-store-{{ .Release.Namespace }}
+  target:
+    name: {{ $serviceName }}-oidc-secrets
+    template:
+      metadata:
+        annotations:
+          resource_kind: '*v1.Secret'
   data:
-    - key: {{ .Values.externalSecrets.oauth2.key }}
-      property: {{ .Values.gloo.authConfig.spec.configs.oauth.clientId | quote }}
-      name: oauth
-
+    - secretKey: oauth
+      remoteRef:
+        key: {{ .Values.externalSecrets.oauth2.key }}
+        property: {{ .Values.gloo.authConfig.spec.configs.oauth.clientId | quote }}
+  refreshInterval: {{ .Values.externalSecrets.service.refreshInterval | default "15m" }}
 {{- end }}
 
 {{- if and .Values.externalSecrets.oauth2.key .Values.gloo.authConfig.spec.configs.clientCredentialsPlugin.config.clientId }}
 ---
-apiVersion: 'kubernetes-client.io/v1'
+apiVersion: 'external-secrets.io/v1beta1'
 kind: ExternalSecret
 metadata:
   name: {{ $serviceName }}-oauth2-client-credentials-secrets
   namespace: {{ .Release.Namespace }}
 spec:
-  backendType: secretsManager
-  {{- if .Values.externalSecrets.service.roleArn }}
-  roleArn: {{ .Values.externalSecrets.service.roleArn }}
-  {{- end }}
+  secretStoreRef:
+    kind: SecretStore
+    name: external-secret-store-{{ .Release.Namespace }}
+  target:
+    name: {{ $serviceName }}-oauth2-client-credentials-secrets
   data:
-    - key: {{ .Values.externalSecrets.oauth2.key }}
-      property: {{ .Values.gloo.authConfig.spec.configs.clientCredentialsPlugin.config.clientId | quote }}
-      name: ClientCredentialsFlow
-
+    - secretKey: ClientCredentialsFlow
+      remoteRef:
+        key: {{ .Values.externalSecrets.oauth2.key }}
+        property: {{ .Values.gloo.authConfig.spec.configs.clientCredentialsPlugin.config.clientId | quote }}
+  refreshInterval: {{ .Values.externalSecrets.service.refreshInterval | default "15m" }}
```

Comment:
Sorry for asking here again. Why didn't you put the default value in the values.yaml? I'd argue that's the idiomatic way and it would also be rendered in the README

Reply:
Hi @crimbler2 ,

We never set pre-defined values in the helm values.yaml files.

@mithie : Should we change the current behaviour and follow the idiomatic way ?

Reply:
I think we're using default values in the values.yaml quite often already. Please correct me if I am wrong, but when I look into this chart's values.yaml I can see a lot of them. I would therefore say to use it that way, yes.

Reply:
You are right. I mixed up. The 15m default is set in values.yaml.